src/core/handle.ml

```diff
@@ -245,7 +257,65 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
       (* Actually add the symbol to the signature. *)
       out 3 "(symb) %s ≔ %a\n" x.elt pp_term t;
       let d = if op then None else Some(t) in
-      let ss = Sig_state.add_symbol ss expo Defin Eager x a impl d in
+      (fst (Sig_state.add_symbol ss expo Defin Eager x a impl d), None)
+  | P_inductive(ms, id, a, l)     ->
+      (* Verify the modifiers. *)
+      let (prop, e, mstrat) = handle_modifiers ms in
+      if prop <> Defin then
+        fatal cmd.pos "Property modifiers cannot be used in inductive types.";
+      if mstrat <> Eager then
+        fatal cmd.pos "Pattern matching strategy modifiers cannot be used \
+                       in inductive types.";
+      (* Add the inductive type in the signature *)
+      let (ss, ind_sym) = handle_symbol ss e Injec Eager id [] a in
+      (* Add the constructors in the signature.  *)
+      let add_cons :
+            sig_state * sym list -> ident * p_term -> sig_state * sym list
+        = fun (ss, cons_list) (id, a) ->
+        let (ss, cons_sym) = handle_symbol ss e Const Eager id [] a in
+        (ss, cons_sym::cons_list)
+      in
+      let (ss, cons_list_rev) = List.fold_left add_cons (ss, []) l in
+      (* Reverse the list of constructors previoulsy computed to preserve the
+         initial order *)
+      let cons_list = List.rev cons_list_rev  in
+      (* Compute the induction principle *)
+      let rec_typ = Inductive.gen_rec_type ss cmd.pos ind_sym cons_list in
+      (* Check the type of the induction principle *)
+      begin
+        match Typing.infer [] rec_typ with
+        | Some _ -> ()
+        | None   ->
+            fatal cmd.pos "The type of the generated inductive principle of
+                           [%a] isn't typable. Please, raise an issue."
+              pp_term rec_typ
+      end;
+      (* Add the induction principle in the signature *)
+      let rec_name = Pos.make cmd.pos (Parser.add_prefix "ind_" id.elt) in
+      if StrSet.mem id.elt !(ss.signature.sign_idents) then
+        begin
+          Sign.add_ident ss.signature rec_name.elt;
+        if !log_enabled then
+          log_hndl "This ident %a is already in the signature \
+                    and now the ident %a too !"
+            Pretty.pp_ident id Pretty.pp_ident rec_name;
+        end;
+      let (ss, rec_sym) =
+        Sig_state.add_symbol ss e Defin Eager rec_name rec_typ [] None
+      in
+      if !log_enabled then
+        log_hndl "The induction principle named %a is %a"
+          Pretty.pp_ident rec_name Print.pp_term rec_typ;
+      (* Compute the rules associated with the induction principle,
+         check the type preservation of the rules and add it to the
+         signature *)
+      let open Stdlib in
+      no_wrn := true;
```

Comment:
We need to reset the value of no_wrn to its previous value. And it is better to make those changes local. So I propose to remove the lines 312, 313 and 318 and instead write:
```
(* in console.ml *)
let with_no_wrn f x =
  let open Stdlib in
  let w = !no_wrn in
  no_wrn := true;
  let y = f x in
  no_wrn := w;
  y

(* here *)
let rs = Inductive.gen_rec_rules ind_sym rec_sym cons_list in
let ss = with_no_wrn (handle_rules ss) rs in
Sign.add_inductive ss.signature ind_sym cons_list rec_sym;
```

Reply:
Oh yes you right ! It's better now. :)